security/pkg/caclient/rotatorfactory_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"testing"
+
+	"istio.io/istio/security/pkg/pki/util/mock"
+)
+
+func TestNewKeyCertBundleRotator(t *testing.T) {
+
+	testCases := map[string]struct {
+		config      *Config
+		expectedErr string
+	}{
+		"null config": {
+			config:      nil,
+			expectedErr: "nil configuration passed",
+		},
+		"onprem env": {
+			config: &Config{
+				CertChainFile: "../platform/testdata/cert-chain-good.pem",
+				Env:           "onprem",
+			},
+			expectedErr: "",
+		},
+		"unspecified env": {
+			config: &Config{
+				CertChainFile: "../platform/testdata/cert-chain-good.pem",
+				Env:           "unspecified",
+			},
+			expectedErr: "",
+		},
+		"Unsupported env": {
+			config: &Config{
+				CertChainFile: "../platform/testdata/cert-chain-good.pem",
+				Env:           "somethig else",
```

Comment:
can we change this to an non-spaced word, e.g. nonexisting-env? It'll be more obvious the string comes from argument, not part of string template itself.

Reply:
Done.